### ISSUE TITLE
Internal Error with SQLAlchemy

### ISSUE THREAD

**Issue by leonarduschen** [python/mypy#12406]
**Crash Report**

Mypy throws internal error using SQLAlchemy plugin - tried using latest development branch.

**Traceback**

- Working directory: `/home/leonarduschen/src/mono/`
- File being evaluated: `./main.py` (details in "To Reproduce" section)

```python
main.py:15: error: INTERNAL ERROR -- Please try using mypy master on Github:
https://mypy.readthedocs.io/en/stable/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 0.950+dev.8427df1630d434ea7e8778f876a0af024b696c69
Traceback (most recent call last):
  File "/home/leonarduschen/src/mono/venv/bin/mypy", line 8, in <module>
    sys.exit(console_entry())
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/__main__.py", line 12, in console_entry
    main(None, sys.stdout, sys.stderr)
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/main.py", line 96, in main
    res, messages, blockers = run_build(sources, options, fscache, t0, stdout, stderr)
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/main.py", line 173, in run_build
    res = build.build(sources, options, None, flush_errors, fscache, stdout, stderr)
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/build.py", line 180, in build
    result = _build(
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/build.py", line 256, in _build
    graph = dispatch(sources, manager, stdout)
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/build.py", line 2733, in dispatch
    process_graph(graph, manager)
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/build.py", line 3077, in process_graph
    process_stale_scc(graph, scc, manager)
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/build.py", line 3169, in process_stale_scc
    mypy.semanal_main.semantic_analysis_for_scc(graph, scc, manager.errors)
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/semanal_main.py", line 78, in semantic_analysis_for_scc
    process_top_levels(graph, scc, patches)
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/semanal_main.py", line 199, in process_top_levels
    deferred, incomplete, progress = semantic_analyze_target(next_id, state,
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/semanal_main.py", line 326, in semantic_analyze_target
    analyzer.refresh_partial(refresh_node,
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/semanal.py", line 409, in refresh_partial
    self.refresh_top_level(node)
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/semanal.py", line 420, in refresh_top_level
    self.accept(d)
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/semanal.py", line 5219, in accept
    node.accept(self)
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/nodes.py", line 1018, in accept
    return visitor.visit_class_def(self)
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/semanal.py", line 1115, in visit_class_def
    self.analyze_class(defn)
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/semanal.py", line 1193, in analyze_class
    self.analyze_class_body_common(defn)
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/semanal.py", line 1202, in analyze_class_body_common
    self.apply_class_plugin_hooks(defn)
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/semanal.py", line 1262, in apply_class_plugin_hooks
    hook(ClassDefContext(defn, base_expr, self))
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/sqlalchemy/ext/mypy/plugin.py", line 258, in _base_cls_hook
    decl_class.scan_declarative_assignments_and_apply_types(ctx.cls, ctx.api)
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/sqlalchemy/ext/mypy/decl_class.py", line 96, in scan_declarative_assignments_and_apply_types
    _scan_declarative_assignment_stmt(
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/sqlalchemy/ext/mypy/decl_class.py", line 468, in _scan_declarative_assignment_stmt
    apply.apply_type_to_mapped_statement(
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/sqlalchemy/ext/mypy/apply.py", line 209, in apply_type_to_mapped_statement
    left_node.type = api.named_type(
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/semanal.py", line 4589, in named_type
    sym = self.lookup_fully_qualified(fullname)
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/semanal.py", line 4555, in lookup_fully_qualified
    ret = self.lookup_fully_qualified_or_none(fullname)
  File "/home/leonarduschen/src/mono/venv/lib/python3.9/site-packages/mypy/semanal.py", line 4571, in lookup_fully_qualified_or_none
    assert '.' in fullname
AssertionError: 
main.py:15: : note: use --pdb to drop into pdb
```

**To Reproduce**

1. Install relevant libraries:
```
mypy                      0.950+dev.8427df1630d434ea7e8778f876a0af024b696c69
mypy-extensions           0.4.3
SQLAlchemy                1.4.27
sqlalchemy2-stubs         0.0.2a19
```

2. Create file named `main.py`
```python
from sqlalchemy import orm, Column, BigInteger

mapper_registry = orm.registry()


class Base(metaclass=orm.decl_api.DeclarativeMeta):
    __abstract__ = True

    registry = mapper_registry
    metadata = mapper_registry.metadata

    __init__ = mapper_registry.constructor


class Person(Base):
    __tablename__ = 'persons'

    id = Column(BigInteger, primary_key=True)
```

3. Run `mypy main.py --show-traceback`

**Your Environment**

- Mypy version used: `mypy 0.950+dev.8427df1630d434ea7e8778f876a0af024b696c69 (compiled: no)`
- Mypy command-line flags: `--show-traceback`
- Mypy configuration options from `mypy.ini` (and other config files):
```
[mypy]  
python_version = 3.9
plugins = sqlalchemy.ext.mypy.plugin
```
- Python version used: `3.9.7`
- Operating system and version: `Pop!_OS 21.10 x86_64`

**Comment by KotlinIsland** [comment]
My first guess is that it's the `sqlalchemy` plugin, can you see it it still happens with it disabled? If that fixes it, try raising it with them.

**Comment by leonarduschen** [comment]
Actually yes, disabling the plugin fixes it indeed, will raise with them

**Comment by leonarduschen** [comment]
Fixed by upgrading SQLAlchemy to 1.4.32
